pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -643,6 +670,7 @@ String getHandlerName() {
     public void redeliverUnacknowledgedMessages() {
         lock.writeLock().lock();
         try {
+            CONSUMER_EPOCH.incrementAndGet(this);
```

Comment:
The `CONSUMER_EPOCH` of `MultiTopicsConsumerImpl` might have a different sequence with `ConsumerImpl` due to the epoch will increase if `ConsumerImpl` reconnects to the topic, here we assume the `MultiTopicsConsumerImpl` and `ConsumerImpl` with same the value.

We can check the message epoch with the epoch of internal consumer?

Reply:
Another point is if users can `MultiTopicsConsumerImpl.redeliverUnacknowledgedMessages()` and `ConsumerImpl.redeliverUnacknowledgedMessages()` will also make the consumer epoch inconsistent.